### PR TITLE
Fix forceFlush() does not clear metrics

### LINKED ISSUE #857: Calling forceFlush() does not clear metrics
**Describe your environment** 

PHP 7.4
Latest commit (https://github.com/open-telemetry/opentelemetry-php/commit/92e7a0d0706f2bf37c4023d19247c2ac6b3511d4)

**Steps to reproduce**

Calling `MeterProvider::forceFlush()` repeatedly will export the same metrics again and again.

```php
<?php

declare(strict_types=1);

require_once __DIR__ . '/vendor/autoload.php';

use OpenTelemetry\SDK\Common\Time\ClockFactory;
use OpenTelemetry\SDK\Metrics\Data\Temporality;
use OpenTelemetry\SDK\Metrics\MeterProviderBuilder;
use OpenTelemetry\SDK\Metrics\MetricExporter\InMemoryExporter;
use OpenTelemetry\SDK\Metrics\MetricReader\ExportingReader;

$exporter = new InMemoryExporter(Temporality::CUMULATIVE);
$reader = new ExportingReader($exporter, ClockFactory::getDefault());
$meterProvider = (new MeterProviderBuilder())->addReader($reader)->build();
$meter = $meterProvider->getMeter('test-meter');
$counter = $meter->createCounter('test-counter');
$counter->add(1);

var_dump(count($exporter->collect())); // 0

$meterProvider->forceFlush();
var_dump(count($exporter->collect())); // 1

$meterProvider->forceFlush();
var_dump(count($exporter->collect())); // 2

$meterProvider->forceFlush();
var_dump(count($exporter->collect())); // 3
```

**What is the expected behavior?**

Calling `MeterProvider::forceFlush()` exports and clears the metrics queue

**What is the actual behavior?**

Calling `MeterProvider::forceFlush()` exports the metrics but does not clear the queue

### PR DESCRIPTION
**Pull request by erdemkose**
Fixes #857

### CODE REVIEW

**Review by bobstrecansky — APPROVED** [review]

File: src/SDK/Metrics/MetricReader/ExportingReader.php [hunk 1]
```diff
@@ -107,6 +107,8 @@ public function forceFlush(): bool
         $collect = $this->doCollect();
         $forceFlush = $this->exporter->forceFlush();
 
+        $this->sources = [];
```
Comment (brettmc) [inline src/SDK/Metrics/MetricReader/ExportingReader.php:110]: I'm not convinced that forceFlush should clear sources, let's see if @Nevay can provide any guidance. Is there something in the spec that says that it should behave that way (or a bug that's bring exhibited that this fixes?)

Comment (Nevay) [reply src/SDK/Metrics/MetricReader/ExportingReader.php:110]: Sources must not be cleared.

Regarding the issue: `InMemoryExporter::collect()` does __not__ reset collected metrics by default, `$exporter->collect(true)` has to be used if collected metrics should be cleared.

Comment (erdemkose) [reply src/SDK/Metrics/MetricReader/ExportingReader.php:110]: I used `InMemoryExporter` for simple reproduction. I have the same issue with `Otlp*Exporter`. With the current implementation, exporters just send the metrics given to them.

**Comment by erdemkose** [comment]
Closing the PR since the behavior is by design. I closed the linked issue, too.

**Pull request closed without merging**